File: android-builder/src/com/mine/c66bridge/ScanReceiver.java
package com.mine.c66bridge;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * Industrial Chainway C66 RFID & Barcode Intent Receiver
 */
public class ScanReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent == null) return;

        String tagData = intent.getStringExtra("scannerdata");
        if (tagData == null) tagData = intent.getStringExtra("barcode");
        if (tagData == null) tagData = intent.getStringExtra("data");
        if (tagData == null) tagData = intent.getStringExtra("barcode_string");
        if (tagData == null) tagData = intent.getStringExtra("tag_id");
        if (tagData == null) tagData = intent.getStringExtra("epc");
        if (tagData == null) tagData = intent.getStringExtra("extra_rfid_data");
        if (tagData == null) tagData = intent.getStringExtra("tag_epc");

        if (tagData == null) {
            byte[] rawBytes = intent.getByteArrayExtra("data");
            if (rawBytes != null && rawBytes.length > 0) {
                tagData = new String(rawBytes).trim();
            }
        }

        if (tagData != null && !tagData.trim().isEmpty()) {
            MainActivity inst = MainActivity.getInstance();
            if (inst != null) {
                inst.dispatchScan(tagData.trim(), "BroadcastIntent: " + intent.getAction());
            }
        }
    }
}

File: android-scanner-bridge/app/src/main/java/com/mine/c66bridge/ScanReceiver.kt
package com.mine.c66bridge

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/**
 * Global Chainway C66 RFID and Barcode Intent Receiver
 */
class ScanReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent?) {
        val action = intent?.action ?: return
        
        val tagData = intent.getStringExtra("scannerdata")
            ?: intent.getStringExtra("barcode")
            ?: intent.getStringExtra("data")
            ?: intent.getStringExtra("barcode_string")
            ?: intent.getStringExtra("tag_id")
            ?: intent.getStringExtra("epc")
            ?: intent.getStringExtra("extra_rfid_data")
            ?: intent.getStringExtra("tag_epc")
            ?: intent.getByteArrayExtra("data")?.let { String(it).trim() }

        if (!tagData.isNullOrEmpty()) {
            MainActivity.instance?.dispatchScanToWeb(tagData.trim())
        }
    }
}
